packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -373,6 +374,16 @@ export function createPatchedFetcher(
             revalidateStore &&
             revalidateStore.revalidate === 0)
 
+        if (
+          hasNoExplicitCacheConfig &&
+          workUnitStore !== undefined &&
+          workUnitStore.type === 'prerender'
+        ) {
+          // If we have no cache config, and we're in Dynamic I/O prerendering, it'll be a dynamic call.
+          // We don't have to issue that dynamic call.
+          return makeHangingPromise(workUnitStore.renderSignal, 'fetch()')
```

Comment:
The hasNoExplicitCacheConfig is not cached and so would cause a regular fetch but we should never actually do that fetch.